File: src/controllers/tokenController.js
var {
    validateData,
    validateDataInsert,
  } = require("../services/tokenServices");
  
  const controlador = {
    listar: {},
  };
  
  controlador.getTokenJira = (req, res) => {
    try {
      validateData()
        .then((resolve) => {
          res.status(200).json({
            status: 200,
            message: "Ok",
            response: resolve,
          });
          console.log(resolve);
        })
        .catch((reject) => {
          res.status(200).json({
            status: 200,
            message: "Ok",
            response: reject,
          });
          console.log(reject);
        });
    } catch (error) {
      res.status(500).json({
        status: 500,
        message: "Internal Server Error",
        response: error,
      });
      console.log(error);
    }
  };
  
  controlador.updateDate = (req, res) => {
    try {
      validateDataInsert(req.body)
        .then((resp) => {
          res.status(200).json({
            status: 200,
            message: "Ok",
            response: resp,
          });
          console.log(resp);
        })
        .catch((err) => {
          res.status(200).json({
            status: 200,
            message: "Ok",
            response: err,
          });
          console.log(err);
        });
    } catch (error) {
      res.status(500).json({
        status: 500,
        message: "Internal Server Error",
        response: error,
      });
      console.log(error);
    }
  };
  
  module.exports = controlador;
  

File: src/controllers/tokenController.py
from flask import jsonify, request
from services.tokenServices import validateData, validateDataInsert


def getTokenJira(req, res):
    try:
        resolve = validateData()
        res.status_code = 200
        return jsonify({
            'status': 200,
            'message': 'Ok',
            'response': resolve
        })
    except Exception as e:
        res.status_code = 500
        return jsonify({
            'status': 500,
            'message': 'Internal Server Error',
            'response': str(e)
        })

def updateDate(req, res):
    try:
        resp = validateDataInsert(req.json)
        res.status_code = 200
        return jsonify({
            'status': 200,
            'message': 'Ok',
            'response': resp
        })
    except Exception as e:
        res.status_code = 500
        return jsonify({
            'status': 500,
            'message': 'Internal Server Error',
            'response': str(e)
        })

# Asignación de funciones al controlador (ya no necesitas esta parte si usas el enfoque de decoradores)
# controlador['getTokenJira'] = get_token_jira
# controlador['updateDate'] = update_date

# Exportación del controlador (también puedes exportar las funciones directamente si usas el enfoque de decoradores)
# return controlador
